### PR TITLE
ENR-32/83 Return branding information on the basis of EC uuid.

### PR DESCRIPTION
**Pull request by asadiqbal08**
@mattdrayer @zubair-arbi Can you guys take a look ? 
[ENT-32](https://openedx.atlassian.net/browse/ENT-32) & [ENT-83](https://openedx.atlassian.net/browse/ENT-83)

### CODE REVIEW

File: enterprise/api.py [hunk 1]
```diff
@@ -7,10 +7,21 @@
 from enterprise.models import EnterpriseCustomerBrandingConfiguration
 
 
-def get_enterprise_branding_info(provider_id=None):
+def get_enterprise_branding_info(**kwargs):
```
Comment (mattdrayer) [inline enterprise/api.py]: Do we really need to open this signature up to such a broad set of use cases?  It seems like below we are only concerned with two parameters -- we should be explicit about the arguments.

File: enterprise/api.py [hunk 2]
```diff
@@ -7,10 +7,21 @@
...
     """
-    return EnterpriseCustomerBrandingConfiguration.objects.filter(
-        enterprise_customer__enterprise_customer_identity_provider__provider_id=provider_id
-    ).first()
+    provider_id = kwargs.get('provider_id', None)
+    enterprise_customer_uuid = kwargs.get('ec_src', None)
+    if provider_id:
+        return EnterpriseCustomerBrandingConfiguration.objects.filter(
+            enterprise_customer__enterprise_customer_identity_provider__provider_id=provider_id
+        ).first()
```
Comment (mattdrayer) [inline enterprise/api.py]: will `.first()` fail if no matches are returned?

File: enterprise/apps.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ class EnterpriseConfig(AppConfig):
 
     name = "enterprise"
     valid_extensions = [".png", ]
-    image_size = 250 * 1024
+    image_size = 4 * 1024
```
Comment (mattdrayer) [inline enterprise/apps.py]: We might want to make this a configurable value for other Open edX operators to manage.

File: enterprise/api.py (hunk 1 above)
Comment (asadiqbal08) [inline enterprise/api.py]: I was thinking the same. Ok I will update it to take two input parameter.

File: enterprise/apps.py (hunk 3 above)
Comment (asadiqbal08) [inline enterprise/apps.py]: It seems me, logo should be enough to have 4kb in size ( not need to change frequently). However we need to scale that logo on client side that will be happened on my other [PR](https://github.com/edx/edx-platform/pull/14064)

**Comment by mattdrayer** [comment]
👍

**Comment by asadiqbal08** [comment]
@haikuginger can you review the change set ? Wana to merge in upcoming day.

**Comment by haikuginger** [comment]
@asadiqbal08, will review today.

File: enterprise/api.py [hunk 4]
```diff
@@ -7,10 +7,19 @@
 from enterprise.models import EnterpriseCustomerBrandingConfiguration
 
 
-def get_enterprise_branding_info(provider_id=None):
+def get_enterprise_branding_info(provider_id=None, ec_uuid=None):
```
Comment (asadiqbal08) [inline enterprise/api.py]: We can also split it into two separate methods in api.
1 - `get_enterprise_branding_info_by_provider_id(provider_id=None)`
2 - `get_enterprise_branding_info_by_ec_uuid(ec_uuid=None)`

Comment (mattdrayer) [inline enterprise/api.py]: I like this idea -- it's good to have very specific functions representing these pieces of system behavior with their own test coverage.

**Comment by asadiqbal08** [comment]
Green Build. 
any yellow thumbs up :)) ? @mattdrayer @haikuginger ^^

File: enterprise/apps.py (hunk 3 above)
Comment (haikuginger) [inline enterprise/apps.py]: I kind of agree with @mattdrayer's initial assessment - 4kB is a bit small of a limit to not be configurable. I'd rather have this pull from the Django settings but have it default to 4kB.

File: enterprise/api.py [hunk 5]
```diff
@@ -7,10 +7,21 @@
 from enterprise.models import EnterpriseCustomerBrandingConfiguration
 
 
-def get_enterprise_branding_info(provider_id=None):
+def get_enterprise_branding_info_by_provider_id(provider_id=None):  # pylint: disable=invalid-name
     """
-    Return the EnterpriseCustomer branding information.
+    Return the EnterpriseCustomer branding information based on enterprise customer provider_id.
+
+    Provider_id: There is 1:1 relation b/w EnterpriseCustomer and SAML Identity provider.
```
Comment (haikuginger) [inline enterprise/api.py]: We can attach to any SSO provider, not just SAML.

File: enterprise/apps.py (hunk 3 above)
Comment (asadiqbal08) [inline enterprise/apps.py]: what is actual required here ? can you give an example ?

File: enterprise/api.py (hunk 5 above)
Comment (saleem-latif) [inline enterprise/api.py]: @asadiqbal08 behavior was updated [here](https://github.com/edx/edx-enterprise/pull/16/files#diff-ce3a6d48e7f42961c07cd219f505d179R27), now we can attach any SSO provider.

**Comment by asadiqbal08** [comment]
@haikuginger kindly take a look.

File: enterprise/api.py [hunk 6]
```diff
@@ -7,10 +7,21 @@
 from enterprise.models import EnterpriseCustomerBrandingConfiguration
 
 
-def get_enterprise_branding_info(provider_id=None):
+def get_enterprise_branding_info_by_provider_id(provider_id=None):  # pylint: disable=invalid-name
     """
-    Return the EnterpriseCustomer branding information.
+    Return the EnterpriseCustomer branding information based on enterprise customer provider_id.
```
Comment (haikuginger) [inline enterprise/api.py]: The provider ID is for the SSO provider, not for the enterprise customer.

File: enterprise/apps.py [hunk 7]
```diff
@@ -17,7 +18,7 @@ class EnterpriseConfig(AppConfig):
 
     name = "enterprise"
     valid_extensions = [".png", ]
-    image_size = 250 * 1024
+    image_size = getattr(settings, 'EC_LOGO_IMAGE_SIZE', 4 * 1024)
```
Comment (haikuginger) [inline enterprise/apps.py]: Nit: `ENTERPRISE_CUSTOMER_LOGO_IMAGE_SIZE`: we're not short on bytes or line space; there's no need to abbreviate. This is going to be a variable that's going to be mixed in with variables for every other  app; we need it to be distinctive.

**Comment by asadiqbal08** [comment]
Done

**Pull request merged**